portfolio.js

```diff
@@ -0,0 +1,54 @@
+const hamburger = document.querySelector('.hamburger');
+const mobileMenu = document.querySelector('.mobile-menu');
+const close = document.querySelector('.close');
+const portfolio = document.querySelector('.portfolio');
+const aboutMe = document.querySelector('.about-me');
+const contact = document.querySelector('.contact');
+
+hamburger.addEventListener('click', () => {
+  const elements = document.querySelectorAll('.logo, .hamburger, .headline, .works, .about, .form');
+
+  elements.forEach((element) => {
+    element.style.display = 'none';
+  });
+
+  mobileMenu.style.display = 'flex';
+  mobileMenu.style.width = '100vw';
+  mobileMenu.style.height = '100vh';
+});
+
+close.addEventListener('click', () => {
+  mobileMenu.style.display = 'none';
+  const elements = document.querySelectorAll('.logo, .hamburger, .headline, .works, .about, .form');
+
+  elements.forEach((element) => {
+    element.style.display = 'flex';
+  });
+});
+
+portfolio.addEventListener('click', () => {
+  mobileMenu.style.display = 'none';
+  const elements = document.querySelectorAll('.logo, .hamburger, .headline, .works, .about, .form');
+
+  elements.forEach((element) => {
+    element.style.display = 'flex';
+  });
+});
+
+aboutMe.addEventListener('click', () => {
+  mobileMenu.style.display = 'none';
+  const elements = document.querySelectorAll('.logo, .hamburger, .headline, .works, .about, .form');
+
+  elements.forEach((element) => {
+    element.style.display = 'flex';
+  });
+});
```

Comment:
- [ ] I see that you are hiding all the content on your page when the menu is open, however, a better approach would be to set the mobile menu container position property as `absolute`, getting it out of the normal document flow, then styling it with 100vh height and 100vw width, covering the entire screen, and then prevent the scrolling in the body, something like this:

```css
.noScroll {
  overflow: hidden
}

.mobile-menu {
  display: none
  position: absolute;
  height: 100vh;
  width: 100vw;
  top: 0;
  left: 0;
}

.mobile-menu.active {
  display: flex
}
``` 

```js
hamburger.addEventListener('click', ()=>{
  mobileMenu.classList.add('active') // This activates the display: flex in CSS.
  document.body.classList.add('noScroll') // This disables the scrolling on the body when the menu is open.
}

close.addEventListener('click', ()=>{
  mobileMenu.classList.remove('active') // This hides the menu
  document.body.classList.remove('noScroll') // This enables the scrolling on the body again.
}

// Also, you can give a common class to each of the mobile menu links (such as menu-link)
// and you can select all of them avoiding repetition

const menuLinks = querySelectorAll('.menu-link')

// Iterate over the resulting array of links:

menuLinks.forEach((link) => {
  link.addEventListener('click', () => {
      mobileMenu.classList.remove('active') // This hides the menu
      document.body.classList.remove('noScroll') // This enables the scrolling on the body again.
  }
}

```

So you don't actually need to hide and show all elements on your page, you can just create this effect by styling your menu overlaying the other elements, and preventing the scrolling, so only the user interaction can close it! Please consider refactoring your code and removing repetition, remember the DRY rule! 🤓 🔧

Reply:
Oh, thank you! That is a more elegant way to do it. I will implement those changes tomorrow.